docs/explaining_ndcube/slicing.rst

```diff
@@ -416,3 +416,110 @@ Let's say we axis order of the ``linewidths`` cube was reversed.
   array([2, 1], dtype=object)
 
 The same result is obtained.
+
+.. _meta_slicing:
+
+Slicing NDMeta
+--------------
+
+Slicing is arguably the most important of `~ndcube.NDMeta`'s functionalities that sets it apart from a simple `dict`.
+It allows axis-aware metadata to be kept consistent with its associated data.
+
+Because `~ndcube.NDMeta` is a subclass of `dict`, the standard Python slicing API is reserved for accessing the values of keys.
+Therefore, slicing is achieved by applying Python's slicing API to `~ndcube.NDMeta`'s `~ndcube.NDMeta.slice` attribute.
+
+.. code-block:: python
+
+  >>> # Instantiate an NDMeta object with axis-aware and axis-agnostic metadata.
+  >>> import astropy.units as u
+  >>> import numpy as np
+  >>> from ndcube import NDMeta
+  >>> raw_meta = {"salutation": "hello", "name": "world",
+  ...             "exposure time": u.Quantity([2] * 4, unit=u.s),
+  ...             "pixel response": np.ones((4, 5))}
+  >>> axes = {"exposure time": 0, "pixel response": (1, 2)}
+  >>> meta = NDMeta(raw_meta, axes=axes, data_shape=(4, 4, 5))
+
+  >>> # Slice NDMeta object.
+  >>> sliced_meta = meta.slice[0, 1:3]
+  >>> sliced_meta.data_shape
+  array([2, 5])
+
+Note that by applying the slice item ``[0, 1:3]`` to ``meta``, the shape of the ``sliced_meta`` has been altered accordingly.
+The first axis has been sliced away, the second has been truncated to a length of 2, and the third remains unchanged.
+The shape of ``"pixel response"`` has been altered:
+
+.. code-block:: python
+
+  >>> sliced_meta["pixel response"].shape
+  (2, 5)
+
+while ``"exposure time"`` has been reduced to a scalar:
+
+.. code-block:: python
+
+  >>> sliced_meta["exposure time"]
+  <Quantity 2. s>
+
+Moreover, because the first axis has been sliced away, ``"exposure time"`` is no longer associated with a data array axis, and so is no longer present in the ``axes`` property:
+
+.. code-block:: python
+
+  >>> list(sliced_meta.axes.keys())
+  ['pixel response']
+
+Finally, note that axis-agnostic metadata is unaltered by the slicing process.
+
+.. code-block:: python
+
+  >>> sliced_meta["salutation"]
+  'hello'
+  >>> sliced_meta["name"]
+  'world'
+
+
+Automatically Slicing NDMeta Attached to Other ND Objects
+*********************************************************
+
+`~ndcube.NDMeta` has a dunder property called ``__ndcube_can_slice__``, which, by default, is set to ``True``.
+When attached to `~ndcube.NDCube`, `~ndcube.NDCubeSequence`, and `~ndcube.NDCollection`, this property lets those classes know that that the metadata object is sliceable.
+Hence, when their ``__getitem__`` methods are called (i.e. when they are sliced), they apply the slice item to ``self.meta.slice`` and attach the result as the meta attribute of the newly sliced ND object.
```

Comment:
This is very developer focused. I think we should skip how this works here and just say "when attached to cube, slicing happens".

Reply:
```suggestion
`~ndcube.NDMeta` has a dunder property called ``__ndcube_can_slice__``, which, by default, is set to ``True``.
When attached to `~ndcube.NDCube`, `~ndcube.NDCubeSequence`, and `~ndcube.NDCollection`, this property lets those classes know to automatically slice the `~ndcube.NDMeta` along with its other sliceable attributes.
```